proto/cosmwasm/wasm/v1/proposal.proto

```diff
@@ -65,8 +65,8 @@ message MigrateContractProposal {
   string contract = 4;
   // CodeID references the new WASM code
   uint64 code_id = 5 [ (gogoproto.customname) = "CodeID" ];
-  // MigrateMsg json encoded message to be passed to the contract on migration
-  bytes migrate_msg = 6;
+  // Msg json encoded message to be passed to the contract on migration
+  bytes msg = 6;
```

Comment:
Nice changes. I was wondering: why is there no ExecuteContractProposal? Would this be useful?

Reply:
Hmmm.. what would it do?

Migrate will override any other auth.

InstantiateProposal is useful if we want to instantiate a code ID which is governance restricted (you can set it so only governance can create new contracts, to avoid eg. everyone making cw20 tokens on a non-DeFi chain)

What would the Execute Proposal do?

All I can think of is to execute a message as an arbitrary account. Which is as powerful as "allow governance to forge a message from arbitrary user". You can make an issue to describe this idea if you think it would be useful and let's wait for Alex to discuss it.